main.cpp

```diff
@@ -53,20 +53,278 @@ Project 4: Part 4 / 9
+    DoubleType& multiply( double rhs );
+    DoubleType& divide( double rhs );
+    operator double() const { return *value; } 
+
+    DoubleType& pow(const FloatType&);
+    DoubleType& pow(const DoubleType&);
+    DoubleType& pow(const IntType&);
+    DoubleType& pow(double dbl);
+
+private:
+    DoubleType& pow_internal(const double exp);
+    double* value = nullptr;
+};
+
+struct FloatType
+{
+    explicit FloatType(float v) : value(new float(v)) {}
+    ~FloatType()
+    {
+        delete value;
+    }
+    FloatType& add( float rhs );
+    FloatType& subtract( float rhs );
+    FloatType& multiply( float rhs );
+    FloatType& divide( float rhs );
+    operator float() const { return *value; }
+
+    FloatType& pow(const FloatType&);
+    FloatType& pow(const DoubleType&);
+    FloatType& pow(const IntType&);
+    FloatType& pow(float flt);
+
+private:
+    FloatType& pow_internal(const float exp);
+    float* value = nullptr;
+};
+
+struct IntType
 {
-    Point& multiply(float m)
+    explicit IntType(int v) : value(new int(v)) {}
+    ~IntType()
     {
-        x *= m;
-        y *= m;
-        return *this;
+        delete value;
     }
+
+    IntType& add( int rhs );
+    IntType& subtract( int rhs );
+    IntType& multiply( int rhs );
+    IntType& divide( int rhs );
+    operator int() const { return *value; }
+    
+    IntType& pow(const FloatType&);
+    IntType& pow(const DoubleType&);
+    IntType& pow(const IntType&);
+    IntType& pow(int i);
+
+private:
+    IntType& pow_internal(const int exp);
+    int* value = nullptr;
+};
+
+DoubleType& DoubleType::add(double rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+DoubleType& DoubleType::subtract(double rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+DoubleType& DoubleType::multiply(double rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+DoubleType& DoubleType::divide(double rhs) 
+{ 
+    if (rhs == 0.0)
+        { std::cout << "warning: floating point division by zero!" << std::endl; }
+    *value /= rhs;
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const FloatType& ft)
+{
+    return pow_internal(static_cast<double>(ft));
+}
+
+DoubleType& DoubleType::pow(const DoubleType& dt)
+{
+    return pow_internal(dt);
+}
+
+DoubleType& DoubleType::pow(const IntType& it)
+{
+    return pow_internal(static_cast<double>(it));
+}
+
+DoubleType& DoubleType::pow(double dbl)
+{
+    return pow_internal(dbl);
+}
+
+DoubleType& DoubleType::pow_internal(const double exp)
+{
+    *value = std::pow(*value, exp);
+    return *this;
+}
+
+FloatType& FloatType::add(float rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+
+FloatType& FloatType::subtract(float rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+
+FloatType& FloatType::multiply(float rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+
+FloatType& FloatType::divide(float rhs) 
+{ 
+    if (rhs == 0.f) 
+        { std::cout << "warning: floating point division by zero!" << std::endl; }
+    *value /= rhs;
```

Comment:
Fix this coding style
Conform to the course coding standard
Curly braces go on their online with no other code. 
- [ ] revice